### DISCUSSION TITLE
G-Code preview source code

### DISCUSSION THREAD

**Discussion by malebuffy** [mriscoc/Ender3V2S1#815] [Q&A]
I cannot find the part of the source code that reads the comments of a g-code  and extracts the thumbnail info? Gcode Preview.h has only a few lines and there seems to be no. Cpp file. Can someone point me to the relevant part of the source code?

**Comment by mriscoc** [comment]
This is my previous version:
https://github.com/mriscoc/Marlin/blob/bugfix-2.1.x/Marlin/src/lcd/e3v2/proui/gcode_preview.cpp

The current is part of ProUIEx library.
